### DISCUSSION TITLE
Consistent ids for fieldArray

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#7157] [Q&A] [answered]
I wonder if there is any form to keep the ids of a field array consistent, I see some re-rendering only because the ids have changed. I've tried to mitigate that using as `key` the index, i.e.: `${name}.${index}` but I still see the re-rendering because of that.

**Comment by fahadsohail482** [comment, marked as answer]
@brun0queiroz field Array has its own id property **id** to use as a key.
Please use that otherwise you'll have some weird behavior. 
![image](https://user-images.githubusercontent.com/46647496/143620248-a0583c1a-5958-4fc4-a406-cb3fb904132f.png)

**Reply by bluebill1049** [reply]
thank you for helping and answering questsions: https://github.com/react-hook-form/react-hook-form/discussions/5340

**Reply by ghost** [reply]
But the issue I have is after every submits the ids are being re-regenerated and triggering the re-rendering only for that, so I wonder if there is a consistent way to generate ids based on some field data.

**Reply by fahadsohail482** [reply]
@brun0queiroz it is not re-rendering you can check this [codesandbox](https://codesandbox.io/s/react-hook-form-usefieldarray-forked-lnebc?file=/src/index.js) from official docs. I've added console here in it to see.
It also have render count on the top.
